internal/npm_install/generate_build_file.js

```diff
@@ -704,6 +711,17 @@ function filterFiles(files, exts = []) {
   // Files with spaces (\x20) or unicode characters (<\x20 && >\x7E) are not allowed in
   // Bazel runfiles. See https://github.com/bazelbuild/bazel/issues/4327
   files = files.filter(f => !f.match(/[^\x21-\x7E]/));
+  files = files.filter(f => {
+    const basename = path.basename(f);
+    // Bazel files of non-nested node_modules would have been renamed earlier with a _ prefix
+    // so we test for both
+    if (/^_?WORKSPACE$/i.test(basename) || /^_?BUILD$/i.test(basename) ||
+        /^_?BUILD\.bazel$/i.test(basename)) {
```

Comment:
Three case insensitive RegExp are probably not particularly efficient.  Something similar to the following may be more efficient and more obvious as to intention.
```
switch (path.basename(f).toUpperCase()) {
  case '_WORKSPACE':
  case 'WORKSPACE':
  case '_BUILD':
  case 'BUILD':
  case '_BUILD.BAZEL':
  case 'BUILD.BAZEL':
    return true;
  default:
    return false;
}
```

Reply:
Good call. I got rid of the regex after some refactoring.